File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        n = len(nums)
        subset = []
        def traverse(i):
            if i >= n: 
                res.append(subset.copy())
                return
            subset.append(nums[i])
            traverse(i+1)
            subset.pop()
            traverse(i+1)
        traverse(0)
        return res

File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
    int n = nums.size();
    int i = n - 2;
    int goal = n - 1;
    for  (int i =n-2;i>=0;i--) {
      int dist = goal - i;
      if (dist <= nums[i]) {
        goal = i;
      }
    }
    return goal == 0;
  }


};
